conan/internal/api/detect_api.py

```diff
@@ -276,7 +276,7 @@ def detect_default_compiler():
         command = cc or cxx
         if "clang" in command.lower():
             return _clang_compiler(command)
-        if "gcc" in command or "g++" in command or "c++" in command:
+        if "cc" in command or "g++" in command or "c++" in command:
```

Comment:
I think this can break some detections of compilers like ``icc``, or ``qcc`` that would suddenly start to be detected as ``gcc`` and that could be breaking.

To solve the issue for Conda, I think it would be better to add ``gnu-c++`` and ``gnu-cc`` to the checks, that would be good enough to detect the Conda one without the risk of affecting other compilers.

Reply:
Makes sense. I think adding `gnu-cc` should do the trick since `c++` would already hit `gnu-c++`. I have made that change.